test/formula.jl

```diff
@@ -1,134 +1,125 @@
-    @test t.terms == [:((&)(x1, x2, x3))]
-    @test t.eterms == [:y, :x1, :x2, :x3]
+    t = @formula(y ~ x1 & x2 & x3)
+    @test t.rhs == x1&x2&x3
+    @test issetequal(terms(t), [y, x1, x2, x3])
 
     ## distributive property of + and &
-    t = Terms(@formula(y ~ x1 & (x2 + x3)))
-    @test t.terms == [:(x1&x2), :(x1&x3)]
-
+    t = @formula(y ~ x1 & (x2 + x3))
+    @test t.rhs == (x1&x2, x1&x3)
+    @test issetequal(terms(t), [y, x1, x2, x3])
+    
     ## ordering of interaction terms is preserved across distributive
-    t = Terms(@formula(y ~ (x2 + x3) & x1))
-    @test t.terms == [:(x2&x1), :(x3&x1)]
+    t = @formula(y ~ (x2 + x3) & x1)
+    @test t.rhs == x2&x1 + x3&x1
 
     ## distributive with *
-    t = Terms(@formula(y ~ (a + b) * c))
-    @test t.terms == [:a, :b, :c, :(a&c), :(b&c)]
+    t = @formula(y ~ (a + b) * c)
+    @test t.rhs == (a, b, c, a&c, b&c)
 
     ## three-way *
-    t = Terms(@formula(y ~ x1 * x2 * x3))
-    @test t.terms == [:x1, :x2, :x3,
-                      :(x1&x2), :(x1&x3), :(x2&x3),
-                      :((&)(x1, x2, x3))]
-    @test t.eterms == [:y, :x1, :x2, :x3]
+    t = @formula(y ~ a * b * c)
+    @test t.rhs == (a, b, c, a&b, a&c, b&c, a&b&c)
+    @test issetequal(terms(t), (y, a, b, c))
 
     ## Interactions with `1` reduce to main effect.
-    t = Terms(@formula(y ~ 1 & x1))
-    @test t.terms == [:x1]
+    t = @formula(y ~ 1 & x1)
+    @test t.rhs == x1
 
-    t = Terms(@formula(y ~ (1 + x1) & x2))
-    @test t.terms == [:x2, :(x1&x2)]
+    t = @formula(y ~ (1 + x1) & x2)
+    @test t.rhs == (x2, x1&x2)
 
     ## PR #54 breaks formula-level equality because original (un-lowered)
     ## expression is kept on Formula struct.  but functional (RHS) equality
     ## should be maintained
-    using StatsModels: dropterm!
+    using StatsModels: drop_term
 
-    @test Terms(dropterm(@formula(foo ~ 1 + bar + baz), :bar)) ==
-        Terms(@formula(foo ~ 1 + baz))
-    @test Terms(dropterm(@formula(foo ~ 1 + bar + baz), 1)) ==
-        Terms(@formula(foo ~ 0 + bar + baz))
+    @test drop_term(@formula(foo ~ 1 + bar + baz), term(:bar)) ==
+        @formula(foo ~ 1 + baz)
+    @test drop_term(@formula(foo ~ 1 + bar + baz), term(1)) ==
+        @formula(foo ~ bar + baz)
 
-    @test_throws ArgumentError dropterm(@formula(foo ~ 0 + bar + baz), 0)
-    @test_throws ArgumentError dropterm(@formula(foo ~ 0 + bar + baz), :boz)
+    # drop_term no longer checks for whether term is found...
+    @test_broken drop_term(@formula(foo ~ bar + baz), term(0))
+    @test_broken drop_term(@formula(foo ~ bar + baz), term(:boz))
 
     form = @formula(foo ~ 1 + bar + baz)
     @test form == @formula(foo ~ 1 + bar + baz)
-    @test Terms(dropterm!(form, :bar)) == Terms(@formula(foo ~ 1 + baz))
-    @test Terms(form) == Terms(@formula(foo ~ 1 + baz))
+    @test drop_term(form, term(:bar)) == @formula(foo ~ 1 + baz)
+    # no longer modifies in place:
+    @test_broken form == @formula(foo ~ 1 + baz)
```

Comment:
This is intended, right? Why not make this a normal test?

Reply:
I think at one point I was trying to be as clear as possible in the tests where behavior had changed by marking previously passing tests that now "failed" as broken.  But in this case I agree it makes sense to test for the new behavior.